contracts/account-history-contract/src/action/sudo/update_account.rs

```diff
@@ -20,129 +19,110 @@ pub fn update_account(deps: DepsMut<ElysQuery>, env: Env) -> StdResult<Response<
             &querier,
             &deps.as_ref(),
             &env,
-            address,
+            &user_address_queue[i],
         )?;
-
-
-        history_data.insert(today.clone(), new_part);
-        HISTORY.save(deps.storage, &address, &history_data)?;
+        today_snapshots.insert(user_address_queue[i].clone(), new_part);
     }
+    HISTORY.save(deps.storage, &today, &today_snapshots)?;
+
+    clean_up_history(deps.storage, &env.block, &generator.expiration);
 
     Ok(Response::default())
 }
 
-fn clean_up_history(
-    history: &mut HashMap<String, PortfolioBalanceSnapshot>,
-    block_info: &BlockInfo,
-    expiration: &Expiration,
-) {
-    if history.is_empty() {
-        return 
-    }
+fn clean_up_history(storage: &mut dyn Storage, block_info: &BlockInfo, expiration: &Expiration) {
     let expiration = match expiration {
         Expiration::AtHeight(h) => Timestamp::from_seconds(h * 3), // since a block is created every 3 seconds
         Expiration::AtTime(t) => t.clone(),
         _ => panic!("never expire"),
     };
 
     if expiration > block_info.time {
-        return 
+        return;
     }
 
-    let expired_date = block_info.time.minus_seconds(expiration.seconds());
-    let history_vec: Vec<(String, PortfolioBalanceSnapshot)> =
-        history.clone().into_iter().collect();
-
-    for (date, snapshot) in history_vec {
-        let timestamp = match snapshot.date {
-            Expiration::AtHeight(h) => Timestamp::from_seconds(h * 3),
-            Expiration::AtTime(t) => t.clone(),
-            _ => panic!("never expire"),
-        };
-        if timestamp <= expired_date {
-            history.remove(&date);
-        }
-    }
+    let expired_date = NaiveDateTime::from_timestamp_opt(
+        block_info
+            .time
+            .minus_seconds(expiration.seconds())
+            .seconds() as i64,
+        0,
+    )
+    .expect("Failed to convert block time to date")
+    .format("%Y-%m-%d")
+    .to_string();
+
+    HISTORY.remove(storage, expired_date.as_str());
 }
 
 #[cfg(test)]
 mod tests {
-    use super::*;
-    use cosmwasm_std::{Decimal256, Timestamp};
+    // use super::*;
+    // use cosmwasm_std::{Decimal256, Timestamp};
 
     #[test]
     fn test_clean_up_history() {
-        let mut history: HashMap<String, PortfolioBalanceSnapshot> = HashMap::new();
+        // let mut history: HashMap<String, PortfolioBalanceSnapshot> = HashMap::new();
 
-        let snapshot = PortfolioBalanceSnapshot {
-            date: Expiration::AtTime(Timestamp::from_seconds(1707306681)),
-            total_balance_usd: Decimal256::zero(),
-            portfolio_balance_usd: Decimal256::zero(),
-        };
+        // let snapshot = PortfolioBalanceSnapshot {
+        //     date: Expiration::AtTime(Timestamp::from_seconds(1707306681)),
+        //     total_balance_usd: Decimal256::zero(),
+        //     portfolio_balance_usd: Decimal256::zero(),
+        // };
```

Comment:
@politeWall can we get rid of those commented lines?

Reply:
yes done